interpreter/interpreter.go

```diff
@@ -372,15 +385,19 @@ func (itrp *Interpreter) doExpression(ctx context.Context, expr semantic.Express
 		if err != nil {
 			return nil, err
 		}
-		if typ := obj.Type().Nature(); typ != semantic.Object {
-			return nil, errors.Newf(codes.Invalid, "cannot access property %q on value of type %s", e.Property, typ)
-		}
-		v, _ := obj.Object().Get(e.Property.Name())
-		if pkg, ok := v.(*Package); ok {
-			// If the property of a member expression represents a package, then the object itself must be a package.
-			return nil, errors.Newf(codes.Invalid, "cannot access imported package %q of imported package %q", pkg.Name(), obj.(*Package).Name())
+
+		if typ := obj.Type().Nature(); typ == semantic.Dynamic {
```

Comment:
I don't think it should. For the standard member expression, we validate that the value is an object and error in the case it isn't.

The idea here was that getting the value of the member of a plain object or one wrapped in a dynamic is the same once you unwrap the dynamic.

I think we'd end up with 2-near identical blocks of code for pulling the value of the member if we move the dynamic check down. This is what motivated me to extract the bulk of the body of the original `semantic.MemberExpression` case as a private func in the first place.

Reply:
Actually, that's a bit dogmatic on my part :cold_sweat:.

Another way to skin this would have been to initialize a separate var for "the object" and then run through the "getter" parts, finally branching on how to handle errors encountered along the way. It wouldn't strictly mean duplicating the whole block, but this way just made a little more sense to me.